app/modules/intelligence/tools/web_tools/code_provider_tool.py

```
@@ -0,0 +1,250 @@
+    def get_public_github_instance(cls):
+        if not cls.gh_token_list:
+            cls.initialize_tokens()
+        token = random.choice(cls.gh_token_list)
+        return Github(token)
+
+    def _get_github_client(self, repo_name: str) -> Github:
+        """Get GitHub client using provider factory."""
+        try:
+            # Use the standard provider factory instead of the GitHub-specific fallback
+            provider = CodeProviderFactory.create_provider()
+            return provider.client
```

Ruff (0.14.4)

Consider moving this statement to an `else` block

(TRY300)